client/python/openlineage/client/transport/http.py

```diff
@@ -0,0 +1,83 @@
+import logging
+from urllib.parse import urljoin
+
+import attr
+
+from typing import Optional
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+from urllib3.util import parse_url
+
+from openlineage.client.run import RunEvent
+from openlineage.client.serde import Serde
+from openlineage.client.transport.transport import Config, Transport
+from openlineage.client.utils import get_only_specified_fields
+
+log = logging.getLogger(__name__)
+
+
+@attr.s
+class HttpConfig(Config):
+    url: str = attr.ib()
+    timeout: float = attr.ib(default=5.0)
+    # check TLS certificates
+    verify: bool = attr.ib(default=True)
+    api_key: Optional[str] = attr.ib(default=None)
```

Comment:
Rather than an api key directly, let's add a `TokenProvider` type. This enables us to add some oauth or other token generator easily in the future. It is also an extension point for users who might want to add their own implementation so they're not blocked on our timeline.

Reply:
I was thinking how to do that. So, we can't have property "token_provider" with `TokenProvider` type if we actually want to both 1) instantiate this config from yaml and 2) provide it as an extension point - since we don't know what types will be instantiated.

What we can do is have something like `auth` dictionary field that has type field that determines what other values are accepted by this config and what class will be instantiated to take care of it. 

So, essentially, a second layer of what we have in this PR in `DefaultTransportFactory` 🙂

Reply:
Added TokenProvider.